### DISCUSSION TITLE
Gradio upgrade

### DISCUSSION THREAD

**Discussion by graz68a** [lllyasviel/Fooocus#3135] [Q&A] [answered]
Hello

When I start Foocus I receive this

_IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade._

When I installed Foocus I didn't install gradio (it was included in the Foocus package I think) . 
Also If I go in the list of installed software I cannot see Gradio.
So how can upgrade gradio ?

I have foocus installed here
I:\fooocus\Fooocus

Thank you
Graziano

**Comment by graz68a** [comment]
ok solved, I have read the warning should be ignored.

**Comment by mashb1t** [comment, marked as answer]
=> https://github.com/lllyasviel/Fooocus/issues/2911#issuecomment-2108362279

**Comment by MokkaBoss1** [comment]
I had a go at upgrading Fooocus gradio version to 5.29

https://github.com/MokkaBoss1/Mokkaboss1_Fooocus/blob/main/README.md

**Reply by mashb1t** [reply]
@MokkaBoss1 looks like you've moved the file to the bin.

**Comment by MokkaBoss1** [comment]
yes I am just uploading a better version it should be uploaded in 1 hour

**Reply by mashb1t** [reply]
why don't you just publish the code to Github (btw. as required by the license of Fooocus)?

**Comment by MokkaBoss1** [comment]
have uploaded  ... good idea about the code will do that aswell

**Comment by MokkaBoss1** [comment]
uploaded the code now as well.
